src/qpid/broker/Selector.cpp

```diff
@@ -108,7 +108,7 @@ MessageSelectorEnv::MessageSelectorEnv(const Message& m) :
 
 const Value MessageSelectorEnv::specialValue(const string& id) const
 {
-    Value v;
+    Value v = Value();
```

Comment:
The original version of this is correct and invokes the default constructor.
The new version is incorrect - it default-constructs a temporary which is copied/moved/elided (depending on compiler version and standard) to v.